spec/tbd_tests_spec.rb

```diff
@@ -2831,6 +2854,153 @@
+    expect(open_roofs.size).to eq(1)
+    open_roof = open_roofs.first
+    roof_id   = open_roof.nameString
+    expect(roof_id).to eq("Level 0 Open area 1 Ceiling Plenum RoofCeiling")
+
+    srr = 0.05
+    gra = TBD.grossRoofArea(model.getSpaces)
+    tm2 = srr * gra
+    rm2 = TBD.addSkyLights(model.getSpaces, {area: tm2})
+    puts TBD.logs unless TBD.logs.empty?
+    expect(TBD.status).to be_zero
+    expect(rm2.round(2)).to eq(gra.round(2))
+
+    entry_skies   = TBD.facets(entry, "Outdoors", "Skylight")
+    office_skies  = TBD.facets(office, "Outdoors", "Skylight")
+    utility_skies = TBD.facets(utility, "Outdoors", "Skylight")
+    open_skies    = TBD.facets(open, "Outdoors", "Skylight")
+
+    expect(entry_skies).to be_empty
+    expect(office_skies).to be_empty
+    expect(utility_skies).to be_empty
+    expect(open_skies.size).to eq(1)
+    open_sky = open_skies.first
+    sky_id   = open_sky.nameString
+    expect(sky_id).to eq("0:0:0:Open area 1:0")
+
+    skm2 = open_sky.grossArea
+    expect((skm2 / rm2).round(2)).to eq(srr)
+
+    # Assign construction to new skylights.
+    construction = TBD.genConstruction(model, {type: :skylight, uo: 2.8})
+    expect(open_sky.setConstruction(construction)).to be true
+    puts TBD.logs unless TBD.logs.empty?
+    expect(TBD.status).to be_zero
+
+    file = File.join(__dir__, "files/osms/out/seb2_sky.osm")
+    model.save(file, true)
+
+    open_well  = open_sky.surface
+    expect(open_well).to_not be_empty
+    open_well  = open_well.get
+    expect(open_well.surfaceType.downcase).to eq("roofceiling")
+    well_id    = open_well.nameString
+    expect(well_id).to eq("0:0:0:Open area 1")
+
+    argh               = {}
+    argh[:option     ] = "regular (BETBG)"
+    argh[:schema_path] = File.join(__dir__, "../tbd.schema.json")
+
+    json     = TBD.process(model, argh)
+    expect(json).to be_a(Hash)
+    expect(json).to have_key(:io)
+    expect(json).to have_key(:surfaces)
+    io       = json[:io      ]
+    surfaces = json[:surfaces]
+    expect(TBD.status).to be_zero
+    expect(TBD.logs).to be_empty
+    expect(surfaces).to be_a(Hash)
+    expect(surfaces.size).to eq(65) # ! 56 before skylight/well/leader lines
+    expect(io).to be_a(Hash)
+    expect(io).to have_key(:edges)
+    expect(io[:edges].size).to eq(115) # ! 106 before skylight/well/leader lines
+
+    # Extra 9 edges:
+    #   - 4x new "skylightjamb" edges
+    #   - 4x new "transition" edges around well
+    #   - 1x "transition" edge along leader line, required for well cutout.
+    sky_jambs = io[:edges].select { |ed| ed[:surfaces].include?(sky_id) }
+    expect(sky_jambs.size).to eq(4)
+
+    sky_jambs.each do |edg|
+      expect(edg[:surfaces].size).to eq(2)
+      expect(edg[:surfaces]).to include(well_id)
+      expect(edg[:type]).to eq(:skylightjamb)
+    end
+
+    roof_edges  = io[:edges].select { |ed| ed[:surfaces].include?(roof_id) }
+    parapets    = roof_edges.select { |ed| ed[:type] == :parapetconvex }
```

Comment:
The initial tests revealed one of the parapet edges was tagged as **:parapet** (and no longer **:parapetconvex**). This would normally be inconsequential (they typically inherit the same PSI factor), yet the change didn't pass the smell test. The issue is linked to a (bad) presumption on Topolys _wires_ and vertex management. See psi.rb changes.